### ISSUE TITLE
Behind the burst compiler

### ISSUE THREAD

**Issue by xoofx** [xoofx/xoofx.github.io#6]
Comments for [Behind the burst compiler](https://xoofx.github.io/blog/2019/03/28/behind-the-burst-compiler/)

**Comment by dwarfland** [comment]
Interesting project! I was wondering if you have had a look at our c# compiler, https://elementscompiler.com. There might be some overlap there, as we already compile c# to native (and other non-il targets); maybe we should talk and possibly cooperate. (Orthogonal to that, it would also be interesting to look at integrating our other languages in their CLR flavor more deeply with Unity, as well). Let me know what you think!

**Comment by nxrighthere** [comment]
Great job Alex, very excited about this feature:
![v](https://user-images.githubusercontent.com/8279330/55726811-1c36f200-5a2a-11e9-9bff-1e7937602d9b.PNG)
Most of my stuff utilizes interop, if the Burst will be able to reduce its cost, it would be just fantastic. My community scream for this feature.

**Comment by nxrighthere** [comment]
Great job Alex, very excited about this feature:
![v](https://user-images.githubusercontent.com/8279330/55726811-1c36f200-5a2a-11e9-9bff-1e7937602d9b.PNG)
Most of my stuff utilizes interop, if the Burst will be able to reduce its cost, it would be just fantastic. My community screaming for this feature.

**Comment by xoofx** [comment]
> very excited about this feature:

I forgot to update the slides, as `[DllImport]` is now supported inside burst jobs

**Comment by nxrighthere** [comment]
@xoofx That's great! We are going to check it out very soon, thanks!

**Comment by xoofx** [comment]
Beware though that some platforms (like iOS) don't allow loading dynamic libraries

**Comment by nxrighthere** [comment]
Yup, they work only in the editor if you target iOS, static libraries are required for builds, but luckily most of the projects that I'm working on is primarily for desktop platforms. (On top of this I'm not a fan to work with constrained environments 😸).

**Comment by RUSshy** [comment]
Will it be open source and available here on github just like we can use CoreRT today ?

**Comment by xoofx** [comment]
> Will it be open source and available here on github just like we can use CoreRT today ?

I would like but It is out of my sole decision. But so far, it looks unlikely though...

**Comment by RUSshy** [comment]
Hopefully it will be released for use outside of unity one day, this is what i want c# to be, focused on AOT, without the extra useless stuff

**Comment by nxrighthere** [comment]
@xoofx Thanks Alex, `[DllImport]` works great. I'm using native libraries with Burst without any problems, however, there's only one problem with the Unity itself which described in [this](https://forum.unity.com/threads/jobs-that-executed-on-the-main-thread-with-run-and-unityengine-api.710042/) thread. It prevents me from mixing UnityEngine API with native code without involving Mono. 😢

**Comment by jamesford42** [comment]
On mobile these slides are not visible, consider having some sort of link to them as a nonembedded document. Or just a better embedding. (This is feedback about the blog which linked here for its comments)
